test/providers.test.js

```diff
@@ -60,6 +59,102 @@ test('json rpc query account', withProvider(async (provider) => {
     expect(response.code_hash).toEqual('11111111111111111111111111111111');
 }));
 
+test('json rpc query view_state', withProvider(async (provider) => {
+    const near = await testUtils.setUpTestConnection();
+    const account = await testUtils.createAccount(near);
+    const contract = await testUtils.deployContract(account, testUtils.generateUniqueString('test'));
+
+    await contract.setValue({ value: 'hello' });
+
+    return testUtils.waitFor(async() => {
+        const response = await provider.query({
+            request_type: 'view_state',
+            finality: 'final',
+            account_id: contract.contractId,
+            prefix_base64: ''
+        });
+        expect(response).toEqual({
+            block_height: expect.any(Number),
+            block_hash: expect.any(String),
+            values: [
+                { key: 'bmFtZQ==', value: 'aGVsbG8=', proof: [] }
+            ],
+            proof: []
+        });
+    });
+}));
+
+test('json rpc query view_account', withProvider(async (provider) => {
+    const response = await provider.query({
+        request_type: 'view_account',
+        finality: 'final',
+        account_id: 'test.near'
+    });
+
+    expect(response).toEqual({
+        block_height: expect.any(Number),
+        block_hash: expect.any(String),
+        amount: expect.any(String),
+        locked: expect.any(String),
+        code_hash: '11111111111111111111111111111111',
+        storage_usage: 182,
+        storage_paid_at: 0,
+    });
+}));
+
+test('json rpc query view_code', withProvider(async (provider) => {
+    const near = await testUtils.setUpTestConnection();
+    const account = await testUtils.createAccount(near);
+    const contract = await testUtils.deployContract(account, testUtils.generateUniqueString('test'));
+
+    return testUtils.waitFor(async() => {
```

Comment:
not clear why `.wairFor` is needed vs jest's native timeout

Reply:
ah, i borrowed the idea from react testing library: https://testing-library.com/docs/dom-testing-library/api-async#waitfor

It waits for the condition to not throw an exception before passing the test. Its like what happens here: https://github.com/near/near-api-js/blob/master/test/providers.test.js#L93.

It seems like the type of thing that will come up repeatedly and i thought this may be a nice general solution.